packages/runtime/test/integration/updateApp.ts

```diff
@@ -172,4 +172,251 @@ describe("Algorand Smart Contracts - Update Application", function () {
+			// create app
+			appID = runtime.deployApp(
+				creator.account,
+				{
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: oldApprovalProgramFileName,
+					clearProgramFilename,
+					...storageConfig,
+					appName: "app3",
+				},
+				{}
+			).appID;
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			// update should be rejected because sender is not creator
+			expectRuntimeError(
+				() =>
+					runtime.updateApp(
+						"app3",
+						creator.address,
+						appID,
+						{
+							metaType: types.MetaType.FILE,
+							approvalProgramFilename: "very-long-approval.teal", // no extra page defined
+							clearProgramFilename: "clear.teal"
+						},
+						{},
+						{}
+					),
+				RUNTIME_ERRORS.TEAL.MAX_LEN_EXCEEDED
+			);
+		});
+
+		it("Should pass: when sufficient extra pages was defined", function () {
+			// create app
+			appID = runtime.deployApp(
+				creator.account,
+				{
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: oldApprovalProgramFileName,
+					clearProgramFilename,
+					...storageConfig,
+					appName: "app4",
+				},
+				{}
+			).appID;
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			// update should be rejected because sender is not creator
+			assert.doesNotThrow(
+				() =>
+					runtime.updateApp(
+						"app4",
+						creator.address,
+						appID,
+						{
+							metaType: types.MetaType.FILE,
+							approvalProgramFilename: "very-long-approval-2-pages.teal",
+							clearProgramFilename: "clear.teal",
+							extraPages: 1 // should pass because total 2 pages needed
+						},
+						{},
+						{}
+					),
+			);
+
+			// verify updated app
+			assert.isDefined(runtime.getApp(appID));
+		});
+
+		it("Should fail: when sufficient extra pages was not defined", function () {
+			// create app
+			appID = runtime.deployApp(
+				creator.account,
+				{
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: oldApprovalProgramFileName,
+					clearProgramFilename,
+					...storageConfig,
+					appName: "app5",
+				},
+				{}
+			).appID;
+			runtime.optInToApp(creator.address, appID, {}, {});
+
+			const app = runtime.getApp(appID);
+			assert.isDefined(app);
+
+			// update should be rejected because sender is not creator
+			expectRuntimeError(
+				() =>
+					runtime.updateApp(
+						"app5",
+						creator.address,
+						appID,
+						{
+							metaType: types.MetaType.FILE,
+							approvalProgramFilename: "very-long-approval-2-pages.teal",
+							clearProgramFilename: "clear.teal",
+							extraPages: 0 // should fail because total 2 pages needed
+						},
+						{},
+						{}
+					),
+				RUNTIME_ERRORS.TEAL.MAX_LEN_EXCEEDED
+			);
+		});
+
+		it.only("Should fail: when sufficient extra pages was not defined", function () {
```

Comment:
please use start using the git hooks. This will not let you commit with the .only

Reply:
done